builder/dockerfile/parser/parser.go

```diff
@@ -148,66 +135,94 @@ func ParseLine(line string, d *Directive) (string, *Node, error) {
 	return "", node, nil
 }
 
+// ParseDirectives attempts to parse a directive from input line
+func ParseDirectives(line string, d *Directives) (err error) {
+	tecMatch := directiveRegexp.FindStringSubmatch(strings.ToLower(line))
+	if len(tecMatch) > 2 {
+		if _, ok := d.usedDirectives[tecMatch[1]]; ok {
+			return fmt.Errorf("only one %v parser directive can be used", tecMatch[1])
+		}
+		defer func() {
+			if err != nil {
+				d.usedDirectives[tecMatch[1]] = struct{}{}
+			}
+		}()
+		switch tecMatch[1] {
+		case "escape":
+			return d.SetEscapeToken(tecMatch[2])
+		}
+	}
+	return errNoDirective
```

Comment:
Is this returned externally? I don't believe we ever returned an error back to the caller on a bad directive, just silently swallowed it (by design).